src/reducer.ts

```diff
@@ -533,7 +660,15 @@ export default produce((draft: State, action: Action) => {
   }
 
   if (action.type === ActionType.LoadMessageArchive) {
-    draft.messages = action.messages
+    draft.messages = action.messages.reduce(
+      (acc, message) => {
+        acc.entities[message.id] = message
+        acc.ids.push(message.id)
+
+        return acc
+      },
+      { entities: {}, ids: [] }
+    )
```

Comment:
This converts from the array of message objects stored in local storage into the "entity state" style, using the current state of messages as the starting point.